src/libs/actions/Report.ts

```diff
@@ -1607,6 +1609,39 @@ function readNewestAction(reportID: string | undefined, shouldResetUnreadMarker
     }
 }
 
+function markAllMessagesAsRead() {
+    const lastReadTime = DateUtils.getDBTimeWithSkew();
+
+    const optimisticData: OnyxUpdate[] = [];
+    const reportIDs: string[] = [];
+    Object.keys(allReports ?? {}).forEach((key: string) => {
+        const report = allReports?.[key];
+        const oneTransactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(
+            report?.reportID,
+            allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report?.reportID}`],
+        );
+        const oneTransactionThreadReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${oneTransactionThreadReportID}`];
+        if (!report || !isUnread(report, oneTransactionThreadReport)) {
+            return;
+        }
```

Comment:
The `report` check should be done at the beginning (before `getOneTransactionThreadReportID`)